app/experimenter/experiments/models/nimbus.py

```diff
@@ -276,7 +276,23 @@ def targeting(self):
 
         if self.locales.count():
             locales = [locale.code for locale in self.locales.all().order_by("code")]
-            expressions.append(f"locale in {locales}")
+
+            if self.application == self.Application.DESKTOP:
+                expressions.append(f"locale in {locales}")
+            else:
+                iso_locale_list = [locale for locale in locales if "-" in locale]
+                iso_locales = set([locale[:2] for locale in iso_locale_list])
+                iso_locales_expression = " || ".join(
+                    [f"'{language}' in locale" for language in list(iso_locales)]
```

Comment:
You don't need to wrap it in a `list()`, `set` also implements an iterator iterface 👍